app/javascript/analytics/dashboard.js

```diff
@@ -223,7 +223,7 @@ function drawMonthCharts({ organizationId, articleId }) {
 function drawInfinityCharts({ organizationId, articleId }) {
   resetActive(document.getElementById('infinity-button'));
   // April 1st is when the DEV analytics feature went into place
-  const beginningOfTime = new Date('2019-4-1');
+  const beginningOfTime = new Date('2019-04-01');
```

Comment:
now that we're at it, could we switch to numbers, `new Date(2019, 3, 1)`?

> Note: Parsing of date strings with the Date constructor (and Date.parse, they are equivalent) is strongly discouraged due to browser differences and inconsistencies.

from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#several_ways_to_create_a_date_object

also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date for details on usage (the month is zero indexed for example)

Reply:
> the month is zero indexed

This is actually the reason I didn't use the numbers[1]. I understand why months are zero-indexed, but given the number of times I've seen devs try to "helpfully" fix a month in a JS date constructor because they think it's a typo, even when there was a comment right above it (I can _especially_ imagine this during Hacktoberfest with all the drive-by PRs), I'm just not sure it's worth it.

If we were accepting user input, I think it might be worth using this constructor instead of trying to fill out an ISO8601/RFC3339 datetime string ourselves. But since the date is harcoded and unlikely to change, and conforms to the standard outlined in the ECMAScript standard linked in the PR description, we're probably in a pretty safe spot.

I'm also wondering if we need this to be a specific date. Do we know what happens if we just use `new Date(0)`?

---

[1] It reminds me of [this slide](https://talks.chastell.net/wrocloverb-2014-lt/#2) from a talk a friend of mine gave a while back:

<img width="1003" alt="new Date(2014, 3, 16) in Java returns April 16, 3914" src="https://user-images.githubusercontent.com/108205/113727627-4c6e0580-96c3-11eb-9348-12b89c4db833.png">

Reply:
We can still add a comment, that will fix everything 🤣 

Can't wait for `Temporal` to land in browsers https://tc39.es/proposal-temporal/docs/index.html

Reply:
ha! [I have been bitten by the zero indexed months so many times](https://twitter.com/s_aitchison/status/1252585923495235584) 🙈

Reply:
lol [same](https://twitter.com/jamie_gaskins/status/622076331895861248)!